Signal/src/ViewControllers/RegistrationViewController.m

```diff
@@ -396,9 +427,65 @@ - (BOOL)textField:(UITextField *)textField
 }
 
 - (BOOL)textFieldShouldReturn:(UITextField *)textField {
-    [self sendCodeAction:nil];
+    [self sendCodeAction];
     [textField resignFirstResponder];
     return NO;
 }
 
+#pragma mark - Debug
+
+#ifdef DEBUG
+
+- (NSString *_Nullable)debugKeychainValueForKey:(NSString *)key
+{
+    OWSCAssert([NSThread isMainThread]);
+    OWSCAssert(key.length > 0);
+
+    NSError *error;
+    NSString *value = [SAMKeychain passwordForService:kKeychainServiceDebug account:key error:&error];
```

Comment:
Take it or leave it:

Why store this in the keychain vs user defaults?

I  feel like the system keychain is a beast to be avoided whenever reasonable.